src/main/kotlin/com/google/androidstudiopoet/ModuleBlueprintFactory.kt

```diff
@@ -24,7 +28,15 @@ object ModuleBlueprintFactory {
             dependency, return methodToCallFromOutside and forget about this module blueprint.
             WARNING: creation of ModuleBlueprint could be expensive
          */
-        return ModuleBlueprint(index, getModuleNameByIndex(index), projectRoot, listOf(), listOf(), config).methodToCallFromOutside
+        synchronized(methodLock[index]) {
```

Comment:
Can you please extract this whole block into a separate method? You can call it something like `getMethodToCallForModule`, then we'll be able to optimize it even more without touching the main method.

Reply:
Thanks for this suggestion, it makes a lot of sense to have it in a function. Hopefully this function will be faster than creating and throwing a ModuleBlueprint once we implement it :p

Reply:
In the long run, I hope we can build modules in the order that doesn't require to have a cache at all unless there is a circular dependency. :)